core/node/libp2p/addrs.go

```diff
@@ -4,24 +4,31 @@ import (
 	"fmt"
 
 	"github.com/libp2p/go-libp2p"
-	host "github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/host"
 	p2pbhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	"github.com/libp2p/go-libp2p/p2p/net/conngater"
 	ma "github.com/multiformats/go-multiaddr"
 	mamask "github.com/whyrusleeping/multiaddr-filter"
 )
 
 func AddrFilters(filters []string) func() (*ma.Filters, Libp2pOpts, error) {
-	return func() (filter *ma.Filters, opts Libp2pOpts, err error) {
-		filter = ma.NewFilters()
-		opts.Opts = append(opts.Opts, libp2p.Filters(filter)) //nolint
+	return func() (*ma.Filters, Libp2pOpts, error) {
+		gater, err := conngater.NewBasicConnectionGater(nil)
+		if err != nil {
+			return nil, Libp2pOpts{}, fmt.Errorf("failed to construct connection gater: %w", err)
+		}
 		for _, s := range filters {
 			f, err := mamask.NewMask(s)
 			if err != nil {
-				return filter, opts, fmt.Errorf("incorrectly formatted address filter in config: %s", s)
+				return nil, Libp2pOpts{}, fmt.Errorf("incorrectly formatted address filter in config: %s", s)
+			}
+			if err := gater.BlockSubnet(f); err != nil {
+				return nil, Libp2pOpts{}, fmt.Errorf("failed to block subnet %s: %w", s, err)
 			}
-			opts.Opts = append(opts.Opts, libp2p.FilterAddresses(f)) //nolint
 		}
-		return filter, opts, nil
+		var opts Libp2pOpts
+		opts.Opts = append(opts.Opts, libp2p.ConnectionGater(gater))
+		return ma.NewFilters(), opts, nil
```

Comment:
This is failing because the returned `ma.Filters` has no impact on the ConnectionGater at all. This means all the commands relying on `n.Filters` to interact with the connection gater are failing.

e.g. https://github.com/ipfs/go-ipfs/blob/7db27bf83a10cd9804a4be8e7421b5ed52bda461/core/commands/swarm.go#L571

Here was the previously used connection gater that exposed that information via the passed in `ma.Filters` https://github.com/libp2p/go-libp2p/blob/v0.14.4/options_filter.go